### PR TITLE
Update HTTPClient, add protobuf library, add audio sinks

### PR DESCRIPTION
**Pull request by kuba2k2**
- HTTPClient's HTTPResponse now extends `ByteStream` directly and provides its full support for chunked streaming, headers, etc.
- protobuf compiler is moved from cspot and used as CMake functions, making it easier to use multiple protobuf sources in complex projects
- audio sinks are moved to bell, as well as the base AudioSink.h
- SPDIFAudioSink is added from @danieltwagner's fork, as it seems he has never started a PR

Breaking changes:
- `ALSAAudioSink` and `PortAudioSink` do not depend on `define`s now, so including them on systems without the libs will break compilation
- audio sinks do not use `std::vector` anymore, instead `uint8_t*` is used

### CODE REVIEW

**Comment by feelfreelinux** [comment]
I am going to hijack this PR to do a larger refactor of bell itself :)

TODO:
- possibly move to nanopb + some simple nanopb C++ wrappers
- use miniaudio.h instead of all the desktop sinks

**Comment by kuba2k2** [comment]
You can also deprecate HTTPStream, as it makes no sense to repeat the same code available in HTTPClient already.

File: include/HTTPClient.h
Comment (feelfreelinux) [inline include/HTTPClient.h:87]: isn't it better to pass it via reference here?

File: include/HTTPClient.h
Comment (feelfreelinux) [inline include/HTTPClient.h:29]: is this used for debugging?

**Review by feelfreelinux — APPROVED** [review]
LGTM, will adjust after merge :)

File: include/HTTPClient.h
Comment (kuba2k2) [reply include/HTTPClient.h:87]: HTTPResponse_t is actually just a unique_ptr so I think it shouldn't really be passed by reference. I added a typedef for this to make method signatures a bit shorter, but maybe naming it differently would be better as currently it can get confused with a struct typedef.

File: include/HTTPClient.h
Comment (kuba2k2) [reply include/HTTPClient.h:29]: It is

**Pull request merged**
